src/Python/qsharp-core/qsharp/loader.py

```diff
@@ -114,6 +127,22 @@ def trace(self, **kwargs) -> Any:
         """
         return qsharp.client.trace(self, **kwargs)
 
+    def as_qir(self) -> pqp.QirModule:
```

Comment:
This was a fairly rough stab on my part for what the final API should look like; before merging to `main`, I think it might make sense to use a `format` keyword argument that allows selecting between `*.ll`, bitcode, and a PyQIR module return values.

Reply:
I'm fairly unfamiliar with the python code for these changes, as I inherited them from previous work. Would you be able to walk me through some of the changes you are thinking about for this sometime?

Reply:
Happy to schedule some time offline, but my thought was an API that's a little closer to that used in https://github.com/qutip/qutip-qip/pull/125.

Reply:
I believe we've now made the needed changes together.